### PR TITLE
Feature/comments

### PR DESCRIPTION
**Pull request by kelum-p**

### CODE REVIEW

File: retro.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
+#Created on 2010-06-15
+#@author: Kelum Peiris (kelum86@gmail.com)
+
+from parsers.CSVParser import CSVParser
+from utils.SQL import SQL
+
+class Group(object):
```
Comment (Coocoolaka) [inline retro.py]: `Group` class

File: retro.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    _headingsType = []
+    _currRow = 0
+
+    def __init__(self, name, fileTypes):
+        self._name = name
+        self._fileTypes = fileTypes
+
+
+    # @param filenameList: List of filenames that contains data.
+    # @summary:
```
Comment (kelum-p) [inline retro.py:22]: Remove comments

File: retro.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    #    Uses the file parsers to parse each data file and combine common data
+    def createGroup(self, filenameList):
```
Comment (kelum-p) [inline retro.py:24]: python method

File: retro.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    try:
+        idx = self._headings.index(heading)
+            self._allData.sort(lambda x, y: cmp(x[idx], y[idx]))
+        except ValueError:
+            print 'Header %s does not exist' % heading
+
+print 'AFTER'
+    for item in self._allData:
+        print item
+
```
Comment (kelum-p) [inline retro.py:64]:
```
def sort(self, heading):
    print 'BEFORE'
        for item in self._allData:
            print item

    try:
        idx = self._headings.index(heading)
            self._allData.sort(lambda x, y: cmp(x[idx], y[idx]))
        except ValueError:
            print 'Header %s does not exist' % heading
```

could be better

File: retro.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+
+
```
Comment (kelum-p) [inline retro.py:67]: one

Comment (kelum-p) [inline retro.py:67]: two

Comment (kelum-p) [inline retro.py:67]: threee

Comment (kelum-p) [inline retro.py:67]: four

Comment (kelum-p) [inline retro.py:67]: five

Comment (kelum-p) [inline retro.py:67]: six

Comment (kelum-p) [inline retro.py:67]: 100

Comment (kelum-p) [inline retro.py:67]: 101

Comment (kelum-p) [inline retro.py:67]: 102

File: retro.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            if item.isdigit() == True:
+                self._headingType.append("LONG")
+            else:
+                self._headingType.append("TEXT")
+
+    self.csvFile.seek(0)
+
+# @param heading: Used to grab the data relevant to the heading
+# @return: Returns the data correspoding to the parameter 'heading'
+def getDataByHeading(self, heading):
```
Comment (kelum-p) [inline retro.py:133]: Better name

Comment (kelum-p) [inline retro.py:133]: 90

Comment (kelum-p) [inline retro.py:133]: 99

Comment (kelum-p) [inline retro.py:133]: 99 problems?

Comment (kelum-p) [inline retro.py:133]: # h1

## h2

**strong**
_italic_

Comment (kelum-p) [inline retro.py:133]: `code` in markdown. `yay`

Comment (kelum-p) [inline retro.py:133]: _emp_

Comment (kelum-p) [inline retro.py:133]: lots of comments

multiple paragraphs

File: retro.py [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+        reader.next() # skip the heading
+
+        for row in enumerate(reader):
+            count += 1
+
+        self.csvFile.seek(0) # Point to the beginning
+
+        return count
+
+def getFilename(self):
```
Comment (kelum-p) [inline retro.py:167]: `getFilename`

Comment (kelum-p) [inline retro.py:167]: one

File: retro.py [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+    def getHeadings(self):
+        return self._headings
+
+    def getHeadingType(self,heading):
+        idx = self._headings.index(heading)
+        return self._headingType[idx]
+
+
+
+
```
Comment (kelum-p) [inline retro.py:180]: Comment on last line

Comment (kelum-p) [inline retro.py:180]: Reply to last line

Comment (kelum-p) [inline retro.py:180]: ee

bb

Comment (kelum-p) [inline retro.py:180]: 10101010

File: retro.py [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+        return count
+
+def getFilename(self):
+    return self._filename
+
+    # @return: return a list of headings
+    def getHeadings(self):
+        return self._headings
+
```
Comment (kelum-p) [inline retro.py:173]: more comments

File: retro.py [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+    def getHeadingType(self,heading):
```
Comment (kelum-p) [inline retro.py:174]: nice comments

Comment (kelum-p) [inline retro.py:174]: ugly comments

File: retro.py [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+        db.commit()
+    db.close()
+
+
+
+
+
+
+
+def sort(self, heading):
```
Comment (kelum-p) [inline retro.py:50]: `sort` method

File: retro.py [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+    print 'BEFORE'
+        for item in self._allData:
+            print item
+
+    try:
```
Comment (kelum-p) [inline retro.py:55]: try

Comment (kelum-p) [inline retro.py:55]: buy

Comment (kelum-p) [inline retro.py:55]: my

Comment (kelum-p) [inline retro.py:55]: guy

Comment (kelum-p) [inline retro.py:55]: sigh

Comment (kelum-p) [inline retro.py:55]: high

Comment (kelum-p) [inline retro.py:55]: rye

File: retro.py (hunk 1 above)
Comment (kelum-p) [inline retro.py:7]: good `group` class

**Pull request merged**